internal/ui/screenshot.go

```diff
@@ -0,0 +1,70 @@
+package ui
+
+import (
+	"image"
+	"image/png"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/container"
+	"fyne.io/fyne/dialog"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/storage"
+	"fyne.io/fyne/widget"
+)
+
+func (l *desktop) screenshot() {
+	bg := l.wm.Capture()
+	l.showCaptureSave(bg)
+}
+
+func (l *desktop) screenshotWindow() {
+	win := l.wm.TopWindow()
+	if win == nil {
+		fyne.LogError("Unable to print window with no window visible", nil)
+		return
+	}
+
+	img := win.Capture()
+	if img == nil {
+		return
+	}
+	l.showCaptureSave(img)
+}
+
+func (l *desktop) showCaptureSave(img image.Image) {
+	w := fyne.CurrentApp().NewWindow("Screenshot")
+	save := widget.NewButton("Save...", func() {
+		saveImage(img, w)
+	})
+	save.Importance = widget.HighImportance
+	buttons := container.NewHBox(
+		layout.NewSpacer(),
+		widget.NewButton("Cancel", func() {
+			w.Close()
+		}),
+		save)
+
+	preview := canvas.NewImageFromImage(img)
+	preview.FillMode = canvas.ImageFillContain
+	w.SetContent(container.NewBorder(nil, buttons, nil, nil, preview))
+	w.Resize(fyne.NewSize(400, 250))
+	w.Show()
+}
+
+func saveImage(pix image.Image, w fyne.Window) {
+	d := dialog.NewFileSave(func(write fyne.URIWriteCloser, err error) {
+		if err != nil {
```

Comment:
You are not checking if `write == nil`. I was noticing that pressing save and then cancelling crashes the desktop.